src/mbgl/style/expression/value.cpp

```diff
@@ -115,6 +121,9 @@ mbgl::Value ValueConverter<mbgl::Value>::fromExpressionValue(const Value& value)
                 array[3],
             };
         },
+        [&](const Collator& collator)->mbgl::Value {
+            return collator.serialize();
```

Comment:
This should also never happen, right?  I think `fromExpressionValue` is only used when handing off the result of an expression for use in the rest of `mbgl::style::*`.  If that's right, we can `assert(false)` here too

Reply:
🤔 What should happen if someone evaluates this expression on its own?

`["collator", { "locale": "en" }]`

It's a useless thing to do, but we should have well defined behavior. If a `Collator` is a kind of `mbgl::Value`, does that mean it needs to be valid to return? In that case we'd need to implement a serialization here?

Reply:
> If a Collator is a kind of mbgl::Value

It's not, though: it's a kind of `mbgl::expression::Value`, but not an `mbgl::Value`.

> What should happen if someone evaluates this expression on its own?

I think my answer to this is basically, "they shouldn't".  Or, rather: they should only do so in a context where they can work with an `mbgl::expression::Value` directly.

Reply:
So I guess ideally it should be a parse error -- if that's the case, then I'm happy with an assert here. Do you know how I can make sure it generates a parse error?

Reply:
I think it shouldn't be _strictly_ a parse error.  For instance, if we wanted to use the `mbgl::style::expression::` subsystem independently for some reason, it's theoretically legitimate to do so. 

It'll already generate a parse error when used in the context of a style property or filter, because we parse those expressions with an expected type based on the property type (or `Boolean` for filters)

Reply:
OK, as long as the "assert" represents a logical error by an implementer, not just invalid input, I'm OK with it... although it makes me uncomfortable that I couldn't really explain the requirements very clearly to someone.

Reply:
Yeah, it's a logical error, but I agree it's a little unclear.  I guess if I had to label the assertion failure, I'd say something like, "fromExpressionValue can't be used for Collator values, because they have no meaningful representation as an `mbgl::Value`"

Also, FWIW, ultimately I feel like we should really replace `mbgl::Value` with `mbgl::expression::Value`, at which point this will be a non-issue